File: app/src/main/java/com/example/a1/NewsActivity.java
package com.example.a1;

import android.content.Intent;
import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.ListView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;

public class NewsActivity extends AppCompatActivity {

    private ListView listViewNews;
    private Button btnBack;

    private ArrayList<NewsItem> newsList;
    private ArrayAdapter<String> adapter;
    private ArrayList<String> newsTitles;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_news);

        listViewNews = findViewById(R.id.listViewNews);
        btnBack = findViewById(R.id.btnBack);

        btnBack.setOnClickListener(v -> finish());

        // Получаем список новостей из хранилища
        newsList = NewsStorage.getNewsList();
        newsTitles = new ArrayList<>();

        // Заполняем список заголовков для отображения
        for (NewsItem item : newsList) {
            newsTitles.add(item.getTitle());
        }

        // Устанавливаем адаптер для ListView
        adapter = new ArrayAdapter<>(this, android.R.layout.simple_list_item_1, newsTitles);
        listViewNews.setAdapter(adapter);

        // Обработка нажатий по элементам списка
        listViewNews.setOnItemClickListener((parent, view, position, id) -> {
            Intent intent = new Intent(NewsActivity.this, FullNewsActivity.class);
            intent.putExtra("news", newsList.get(position)); // newsList.get(position) должен быть Serializable
            startActivity(intent);
        });
    }
}

File: app/src/main/java/com/example/a1/NewsActivity.kt
package com.example.a1

import android.content.Intent
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.AdapterView.OnItemClickListener
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import com.example.a1.SupabaseClient.getNewsTitles

class NewsActivity : AppCompatActivity() {
    private var listViewNews: ListView? = null
    private var btnBack: Button? = null
    private var newsTitles: ArrayList<String>? = null
    private var adapter: ArrayAdapter<String>? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_news)

        listViewNews = findViewById(R.id.listViewNews)
        btnBack = findViewById(R.id.btnBack)

        btnBack.setOnClickListener(View.OnClickListener { v: View? -> finish() })

        // Получение списка новостей из Supabase
        Thread {
            newsTitles = getNewsTitles()
            // Обновление UI — только в главном потоке!
            runOnUiThread {
                adapter =
                    ArrayAdapter(
                        this,
                        android.R.layout.simple_list_item_1,
                        newsTitles!!
                    )
                listViewNews.setAdapter(adapter)
                listViewNews.setOnItemClickListener(OnItemClickListener { parent: AdapterView<*>?, view: View?, position: Int, id: Long ->
                    val intent = Intent(
                        this@NewsActivity,
                        FullNewsActivity::class.java
                    )
                    intent.putExtra("newsTitle", newsTitles!![position])
                    startActivity(intent)
                })
            }
        }.start()
    }
}